kornia/contrib/extract_patches.py

```diff
@@ -78,45 +85,19 @@ def __init__(
         self.stride: Tuple[int, int] = _pair(stride)
         self.padding: Tuple[int, int] = _pair(padding)
 
-        # create base kernel
-        self.kernel: torch.Tensor = self.create_kernel(self.window_size)
-
-    @staticmethod
-    def create_kernel(
-            window_size: Tuple[int, int]) -> torch.Tensor:
-        r"""Creates a binary kernel to extract the patches. If the window size
-        is HxW will create a (H*W)xHxW kernel.
-        """
-        window_range: int = window_size[0] * window_size[1]
-        kernel: torch.Tensor = torch.zeros(window_range, window_range)
-        for i in range(window_range):
-            kernel[i, i] += 1.0
-        return kernel.view(window_range, 1, window_size[0], window_size[1])
-
```

Comment:
since we are make stateless function probably it's better to implement the function as functional and the `nn.Module` wrap around it. In general we try to follow this last coding pattern for stateless functions. What do you think ?

Reply:
Yep, I agree with that. I looked around and `contrib.MaxBlurPool2d` is using the same pattern.

Reply:
indeed. It should be adapted too